src/hooks/data/preview/useAssetToYbToken.ts

```diff
@@ -0,0 +1,87 @@
+import { Address } from "wagmi"
+
+import { VaultType } from "@/lib/types"
+import { useBalancerAssetToYbToken } from "@/hooks/data/preview/useBalancerAssetToYbToken"
+import { useCurveAssetToYbToken } from "@/hooks/data/preview/useCurveAssetToYbToken"
+import { useTokenAssetToYbToken } from "@/hooks/data/preview/useTokenAssetToYbToken"
+import {
+  useIsCurveCompounder,
+  useIsTokenCompounder,
+} from "@/hooks/useVaultTypes"
+
+import { useTxSettings } from "@/store/txSettings"
+
+import { DEFAULT_SLIPPAGE } from "@/constant/env"
+
+export function useAssetToYbToken({
+  chainId,
+  id,
+  token = "0x",
+  amount,
+  type,
+}: {
+  chainId: number
+  id: number | undefined
+  token: Address | undefined
+  amount: string
+  type: VaultType
+}) {
+  const isCurve = useIsCurveCompounder(type)
+  const isToken = useIsTokenCompounder(type)
+
+  const txSettings = useTxSettings()
+  const slippage = parseFloat(
+    txSettings.isSlippageAuto
```

Comment:
This should probably be abstracted so we don't need to manually perform this logic everywhere we are using slippage tolerance.

You could add a hook. Something like this:

```ts
export const useSlippageSetting = () => {
  const [isSlippageAuto, slippageTolerance] = useTxSettings((state) => [
    state.isSlippageAuto,
    state.slippageTolerance,
  ])
  return isSlippageAuto ? DEFAULT_SLIPPAGE : slippageTolerance
}
```

Reply:
yes